rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,328 @@
+This RFC describes the document types containing the information required to form this registry.
+It uses [RFC004](rfc004-distributed-document-format.md) as underlying data format. The origin and requirements of
+certificates are defined by RFC008.
+
+## 2. Terminology
+
+* **Document**: a distributed document as specified by [RFC004](rfc004-distributed-document-format.md).
+* **Organization**: a care organization exchanging data with other care organizations over a Nuts network.
+* **Vendor**: an object developing software for care organizations participating in a Nuts network.
+  Organizations access the Nuts network through their vendor's software.
+* **Endpoint**: a URI or URL exposed by a vendor which can be used by other vendors to pull data.
+* **Service**: a group of endpoints implementing a service required by a Bolt.
+* **Object**: an instance of a vendor, organization or endpoint.
+
+Other terminology comes from the [RFC001 Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Domain Model
+The following UML entity diagram displays how vendors, organizations, endpoints and services relate:
+
+![RFC006 Domain Model](../.gitbook/assets/rfc006-registry-domain-model.svg)
+
+## 4. Format
+Entities MUST be represented as JSON document as the JWS payload defined by [RFC004](rfc004-distributed-document-format.md).
+The **cty** header parameter from [RFC004](rfc004-distributed-document-format.md)) MUST contain the type defined for
+that particular object type. The type parameter `o` MUST be used to indicate creation or an update, e.g:
+
+`registry/vendor;o=update`
+
+Other values than `update` or `create` for the `o` parameter SHALL NOT be used.
+
+## 4.1. Creating and updating
+When creating entities its required fields MUST be specified as JSON document. The `o` parameter MUST be specified as `create`.
+
+To update an object's fields [JSON Patch (RFC6902)](https://tools.ietf.org/html/rfc6902) is used. The object to update
+MUST be specified by specifying the **tid** (*timeline ID*, see [RFC004](rfc004-distributed-document-format.md))
+The `o` parameter MUST be specified as `update`. An update document MUST contain the following fields:
+* **patch** MUST contain an array with patch operations according to [RFC6902](https://tools.ietf.org/html/rfc6902).
+
+If one of the operations fails the document MUST be rejected.
+
+Example for document type `registry/organization;o=update`:
+```json
+{
+    "patch": [
+       { "op": "replace", "path": "/name", "value": "NoPlaceLikeHome Thuiszorg" }
+    ]
+}
+```
+
+When using a destructive operation (especially when using `remove` or `replace` on arrays) vendors COULD use the
+`test` operation to make sure the intended values are removed.
+
+## 4.2. Signing
+All documents MUST be signed by the vendor who owns the object (be it a vendor, organization or endpoint). The only
```

Comment:
"Ownership" is a concept; when a party signs and publishes a document, that party owns it. Thus if an organization or endpoint is published, it is owned (and can only be updated) by the publishing party. If a vendor adds a new CA certificate, ownership is 'transferred' or 'shared' with that new CA certificate and it (or rather; signing certificates issued under that CA certificate) can be used to update the document.

Now I think of it, we register the CA certificate which can be rotated, so we need a concept (but is probably internal to the software) of 'entity' or 'party' (the vendor) which is authenticated by one or more CA certificates. Then an entity 'owns' a set of documents rather than a specific CA certificate. When a new CA certificate is registered it's added to the cert. set of that entity. Ownership can be transferred by simply adding the new certificates and removing the old.

Reply:
That's true

Reply:
Decision: when a vendor re-registers itself (because it has lost its CA and signing keys) it must do so under the same ID (with a new vendor CA certificate and proof). From that point in time (or rather, in DAG) the new CA certificate is added to the pool of CA certificates that authenticate the vendor, meaning it can update documents that were originally published under the old CA certificate.